assignment1_calculator.c

```diff
@@ -0,0 +1,125 @@
+// Kalpvriksha Program | Batch-2025: C Programming Assignments covering Calculator.
+
+// assignment 1: Calculator, input format :- 3+5
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+#define MAX_EXPR_LEN 100
+
+typedef enum {
+    NO_ERROR,
+    DIVISION_BY_ZERO,
+    INVALID_EXPRESSION
+} ErrorType;
+
+ErrorType error = NO_ERROR;
+
+char* skip_whitespace(char *expr) {
+    while (*expr == ' ') expr++;
+    return expr;
+}
+
+int evaluate_term(char **expr);
+
+int evaluate_expression(char **expr);
+
+int parse_integer(char **expr) {
+    *expr = skip_whitespace(*expr);
+    int result = 0;
+    while (isdigit(**expr)) {
+        result = result * 10 + (**expr - '0');
+        (*expr)++;
+    }
+    return result;
+}
+
+int evaluate_factor(char **expr) {
+    *expr = skip_whitespace(*expr);
+    if (isdigit(**expr)) {
+        return parse_integer(expr);
+    } else {
+        error = INVALID_EXPRESSION;
+        return 0;
+    }
+}
+
+int evaluate_term(char **expr) {
+    int value = evaluate_factor(expr);
+
+    while (**expr == '*' || **expr == '/') {
+        char op = **expr;
```

Comment:
Can variable names match below standards, take care of this at all places:

Use descriptive names: Choose meaningful names that reflect the purpose of the variable.
Use Camelcase: Use lowercase letters with underscores to separate words.
ex :
int totalCount;
float averageScore;
char studentName[50];
Global variables:
ex :
int totalCount;

Constants:

Use all uppercase letters with underscores to separate words.
ex:
#define MAX_BUFFER_SIZE 1024
const float PI = 3.14159